include/seqan3/argument_parser/validators.hpp

```diff
@@ -297,68 +286,354 @@ class file_ext_validator
+ * \ingroup argument_parser
+ * \implements seqan3::validator_concept
+ *
+ * \details
+ *
+ * The struct acts as a functor that throws a seqan3::parser_invalid_argument exception whenever a given path
+ * (std::filesystem::path) is not writable. This can happen if either the parent path does not exists, or the
+ * path doesn't have the proper write permissions.
+ *
+ * \snippet test/snippet/argument_parser/validators_output_directory.cpp usage
+ *
+ * \note The validator works on every type that can be implicitly converted to std::filesystem::path.
+ */
+class output_directory_validator : public file_validator_base
+{
+public:
+    // Imported from base class.
+    using file_validator_base::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
      */
-    template <std::ranges::ForwardRange range_type>
-    //!\cond
-        requires std::ConvertibleTo<value_type_t<range_type>, std::filesystem::path const &>
-    //!\endcond
-    void operator()(range_type const & v) const
+    output_directory_validator() = default;                                               //!< Defaulted.
+    output_directory_validator(output_directory_validator const &) = default;             //!< Defaulted.
+    output_directory_validator(output_directory_validator &&) = default;                  //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator const &) = default; //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator &&) = default;      //!< Defaulted.
+    virtual ~output_directory_validator() = default;                                      //!< Virtual Destructor.
+
+    // Import base constructor.
+    using file_validator_base::file_validator_base;
+    //!\}
+
+    // Import the base::operator().
+    using file_validator_base::operator();
+
+    /*!\brief Tests whether path is writable.
+     * \param dir The input value to check.
+     * \throws parser_invalid_argument if the validation process failed, or
+     *         std::filesystem::filesystem_error on underlying OS API errors.
+     */
+    virtual void operator()(std::filesystem::path const & dir) const override
     {
-         std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+        bool dir_exists = std::filesystem::exists(dir);
+        // Make sure the created dir is deleted after we are done.
+        std::error_code ec;
+        std::filesystem::create_directory(dir, ec); // does nothing and is not treated as error if path already exists.
+        // if error code was set or if dummy.txt could not be created within the output dir, throw an error.
+        if (static_cast<bool>(ec))
+            throw parser_invalid_argument(detail::to_string("Cannot create directory: ", dir, "!"));
+
+        if (!dir_exists)
+        {
+            detail::safe_filesystem_entry dir_guard{dir};
+            filename_is_writable(dir / "dummy.txt");
+            dir_guard.remove_all();
```

Comment:
yes, it is, but in this case if it fails (which is very unlikely) it will throw. In the destructor it will trigger an assertion but only in debug mode. So in case something went wrong on that IO level it will  have a more understandable error.